packages/runtime/src/runtime.ts

```diff
@@ -62,55 +63,67 @@ export class Runtime {
     return a;
   }
 
+  /**
+   * asserts if account address is defined
+   * @param addr account address
+   * @param line line number in TEAL file
+   * Note: if user is accessing this function directly through runtime,
+   * the line number is unknown
+   */
+  assertAddressDefined (addr: string | undefined, line?: number): string {
+    if (addr === undefined) {
+      throw new TealError(ERRORS.TEAL.ACCOUNT_DOES_NOT_EXIST, { line: line });
+    }
+    return addr;
+  }
+
   /**
    * asserts if application exists in state
+   * @param app application
    * @param appId application index
    * @param line line number in TEAL file
    * Note: if user is accessing this function directly through runtime,
    * the line number is unknown
    */
-  assertAppDefined (appId: number, line?: number): SSCAttributesM {
+  assertAppDefined (app: SSCAttributesM | undefined, appId: number, line?: number): SSCAttributesM {
     const lineNumber = line ?? 'unknown';
-    const app = this.ctx.state.globalApps.get(appId);
     if (app === undefined) {
       throw new TealError(ERRORS.TEAL.APP_NOT_FOUND, { appId: appId, line: lineNumber });
     }
     return app;
   }
 
   /**
-   * Fetch account using accountIndex from `Accounts` list
-   * Accounts: List of accounts in addition to the sender
-   * that may be accessed from the application's approval-program and clear-state-program.
-   * @param accountIndex index of account to fetch from account list
-   * @param line line number
-   * NOTE: index 0 represents txn sender account
-   * - if user is accessing this function directly through runtime, the line number is unknown
+   * Fetches app from `this.store`
+   * @param appId Application Index
    */
-  getAccount (accountIndex: bigint, line?: number): StoreAccountI {
-    let account: StoreAccountI | undefined;
-    if (accountIndex === BIGINT0) {
-      const senderAccount = encodeAddress(this.ctx.tx.snd);
-      account = this.ctx.state.accounts.get(senderAccount);
-    } else {
-      const accIndex = accountIndex - BIGINT1;
-      checkIndexBound(Number(accIndex), this.ctx.tx.apat, line);
-      const pkBuffer = this.ctx.tx.apat[Number(accIndex)];
-      account = this.ctx.state.accounts.get(encodeAddress(pkBuffer));
+  getApp (appId: number): SSCAttributesM {
+    if (!this.store.globalApps.has(appId)) {
+      throw new TealError(ERRORS.TEAL.APP_NOT_FOUND, { appId: appId, line: 'unknown' });
     }
+    const accAddress = this.assertAddressDefined(this.store.globalApps.get(appId));
+    const account = this.store.accounts.get(accAddress);
+    const app = account?.createdApps.get(appId);
+    return this.assertAppDefined(app, appId);
```

Comment:
Please create a function `getApp` in `StoreAccount`. We should operate on an account attribute (`createdApps` in this case) in account only. So `runtime.getApp() => calls account.getApp()` which get's the app from account's global state.

Reply:
yes